runtime/gc_modron_standard/ReadBarrierVerifier.cpp

```diff
@@ -0,0 +1,241 @@
+/*******************************************************************************
+ * Copyright (c) 1991, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+
+/**
+ * @file
+ * @ingroup GC_Modron_Base
+ */
+
+#if defined(OMR_ENV_DATA64) && !defined(OMR_GC_COMPRESSED_POINTERS)
+
+#include "ReadBarrierVerifier.hpp"
+
+#include "AtomicOperations.hpp"
+#include "Debug.hpp"
+#include "EnvironmentStandard.hpp"
+#include "HeapRegionManager.hpp"
+#include "j9cfg.h"
+#include "mmhook_internal.h"
+#include "ModronAssertions.h"
+
+#include "RootScannerReadBarrierVerifier.hpp"
+#include "Scavenger.hpp"
+#include "SublistFragment.hpp"
+
+MM_ReadBarrierVerifier *
+MM_ReadBarrierVerifier::newInstance(MM_EnvironmentBase *env)
+{
+	MM_ReadBarrierVerifier *barrier;
+
+	barrier = (MM_ReadBarrierVerifier *)env->getForge()->allocate(sizeof(MM_ReadBarrierVerifier), MM_AllocationCategory::FIXED, J9_GET_CALLSITE());
+	if (barrier) {
+		new(barrier) MM_ReadBarrierVerifier(env);
+		if (!barrier->initialize(env)) {
+			barrier->kill(env);
+			barrier = NULL;
+		}
+	}
+
+	return barrier;
+}
+
+void 
+MM_ReadBarrierVerifier::kill(MM_EnvironmentBase *env)
+{
+	tearDown(env);
+	env->getForge()->free(this);
+}
+
+bool
+MM_ReadBarrierVerifier::preObjectRead(J9VMThread *vmThread, J9Object *srcObject, fj9object_t *srcAddress)
+{
```

Comment:
In general these virtual methods should call parent's functionality as well. It happen to be no need for now because of limited scope. "As is" is ok for now, but we should not forget about it